frame/pablo/src/stable_swap.rs

```diff
@@ -0,0 +1,240 @@
+use std::marker::PhantomData;
+
+use crate::{Config, Error, PoolConfiguration, PoolCount, Pools};
+use composable_maths::dex::stable_swap::{compute_base, compute_d};
+use composable_traits::{
+	currency::{CurrencyFactory, RangeId},
+	defi::CurrencyPair,
+	dex::StableSwapPoolInfo,
+	math::{safe_multiply_by_rational, SafeAdd, SafeSub},
+};
+use frame_support::{
+	pallet_prelude::*,
+	traits::fungibles::{Inspect, Mutate, Transfer},
+};
+use sp_runtime::{
+	traits::{CheckedAdd, Convert, One, Zero},
+	ArithmeticError, DispatchError, Permill,
+};
+
+use sp_std::ops::Mul;
+
+pub(crate) struct StableSwap<T>(PhantomData<T>);
+
+impl<T: Config> StableSwap<T> {
+	pub fn do_create_pool(
```

Comment:
Should these functions be
```suggestion
        [#transactional]
	pub fn do_create_pool(
```
? Just asking, because I'm not sure about the behavior of this annotation in case of chained calls to functions that modify storage

Reply:
As stated, it is transactional, meaning that it save the current storage state, execute the function and rollback if Err is returned

Reply:
do_create_pool() in pablo/src/lib.rs has #[transactional] which calls this function so I think suggested change is not required.

Reply:
@hussein-aitlahcen storage transaction part was clear, just that I was curious what the behavior was when there is a chained call. 
@vivekvpandya seems like the way it should work, but I wasn't sure.